File: organization-service/src/main/java/github/Jimoou/organizationservice/Entity/Organization.java
package github.Jimoou.organizationservice.Entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "organizations")
public class Organization {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false)
  private String organizationName;

  private String organizationDescription;

  @Column(nullable = false, unique = true)
  private String organizationCode;

  @CreationTimestamp
  private LocalDateTime createdDate;
}

File: react-frontend/src/model/Organization.ts
class Organization {
  id: number;
  organizationName: string;
  organizationDescription: string;
  organizationCode: string;
  createdDate: string;
  constructor(
    id: number,
    organizationName: string,
    organizationDescription: string,
    organizationCode: string,
    createdDate: string
  ) {
    this.id = id;
    this.organizationName = organizationName;
    this.organizationCode = organizationCode;
    this.organizationDescription = organizationDescription;
    this.createdDate = createdDate;
  }
}
export default Organization;
